app/src/main/java/org/oppia/android/app/settings/profile/ProfileEditActivityPresenter.kt

```diff
@@ -74,6 +80,15 @@ class ProfileEditActivityPresenter @Inject constructor(
       }
     )
 
+    profileEditViewModel.isProfileDeletionDialogShown.observe(
+      activity,
+      Observer {
+        if (it) {
+          showDeletionDialog(profileId)
+        }
+      }
+    )
+
```

Comment:
@prathamk22 Can we try using `onRestoreInstance` instead

Reply:
Yes we can use that. But After watching the commit history. I found that previously we were using onRestoreInstance to update the title and that was removed and was declared in the viewmodel class. So I thought I should go with the flow and should use the ViewModel instead. Should I use onRestoreInstance again in this?

Reply:
That's great, leave it as it is